### Predictive Modeling
__Bima Triadi Ruslan__
05211740000065 

kelas Penggalian Data

Algoritma yang digunakan Naive Bayes

## Langkah-langkah pengerjaan kuis ini terdiri dari :
    - Import Libraries
    - Load dataset
    - Praproses data
    - Pembagian dataset menjadi data training dengan data testing
    - Membangun model prediksi
    - Melatih model menggunakan data training
    - Evaluasi kinerja model

## Langkah import Libraries

In [1]:
import matplotlib.pyplot as plt
import seaborn as sns
import pandas as pd

In [2]:
path = r"D:\KULIAH\CODING\PDT\Kuis" + "\\"
file = "loan_data.csv"

In [3]:
data_utama = pd.read_csv(path + file)
data_utama.corr()

ApplicantIncome  CoapplicantIncome  LoanAmount  \
ApplicantIncome           1.000000          -0.116605    0.570909   
CoapplicantIncome        -0.116605           1.000000    0.188619   
LoanAmount                0.570909           0.188619    1.000000   
Loan_Amount_Term         -0.045306          -0.059878    0.039447   
Credit_History           -0.014715          -0.002056   -0.008433   

                   Loan_Amount_Term  Credit_History  
ApplicantIncome           -0.045306       -0.014715  
CoapplicantIncome         -0.059878       -0.002056  
LoanAmount                 0.039447       -0.008433  
Loan_Amount_Term           1.000000        0.001470  
Credit_History             0.001470        1.000000

## Langkah Praproses Data

In [4]:
data_utama.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 614 entries, 0 to 613
Data columns (total 13 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Loan_ID            614 non-null    object 
 1   Gender             601 non-null    object 
 2   Married            611 non-null    object 
 3   Dependents         599 non-null    object 
 4   Education          614 non-null    object 
 5   Self_Employed      582 non-null    object 
 6   ApplicantIncome    614 non-null    int64  
 7   CoapplicantIncome  614 non-null    float64
 8   LoanAmount         592 non-null    float64
 9   Loan_Amount_Term   600 non-null    float64
 10  Credit_History     564 non-null    float64
 11  Property_Area      614 non-null    object 
 12  Loan_Status        614 non-null    object 
dtypes: float64(4), int64(1), object(8)
memory usage: 62.5+ KB


Langkah pertama dari praproses data adalah menggunakan function _.info()_ untuk memahami dataset ini terdiri dari kolom apa saja dengan tipe data apa saja.

### Penjelasan detail dari tiap kolom :
    - Loan_ID             = Kode pinjaman
    - Gender              = Jenis Kelamin (Male, Female)
    - Married             = Status Perkawainan(Yes, No)
    - Dependents          = Jumlah Tanggungan Keluarga (0 –3)
    - Education           = Status Pendidikan (Graduate, Not Graduate)
    - Self_Employed       = Status Pekerjaan (Yes, No)
    - ApplicantIncome     = Penghasilan Pelamarper tahun(USD)
    - CoapplicantIncome   = Penghasilan Pasangan Pelamar per tahun (USD)
    - LoanAmount          = Jumlah Pinjaman (Ribuan USD)
    - Loan_Amount_Term    = Jangka Waktu Angsuran Pinjaman (hari)
    - Credit_History      = Catatan apakah pernah melakukan pinjaman sebelumnya (0: Tdk, 1: Ya)
    - Property_Area       = Daerah lokasi rumah pelamar Urban: Perkotaan, Semiurban: Semi perkotaan, Rural: Pedesaan)
    - Loan_Status         = Status Persetujuan Pinjaman (1: Disetujui, 0: Ditolak)

In [5]:
data_copy = data_utama.copy()
data_copy

Loan_ID  Gender Married Dependents     Education Self_Employed  \
0    LP001002    Male      No          0      Graduate            No   
1    LP001003    Male     Yes          1      Graduate            No   
2    LP001005    Male     Yes          0      Graduate           Yes   
3    LP001006    Male     Yes          0  Not Graduate            No   
4    LP001008    Male      No          0      Graduate            No   
..        ...     ...     ...        ...           ...           ...   
609  LP002978  Female      No          0      Graduate            No   
610  LP002979    Male     Yes         3+      Graduate            No   
611  LP002983    Male     Yes          1      Graduate            No   
612  LP002984    Male     Yes          2      Graduate            No   
613  LP002990  Female      No          0      Graduate           Yes   

     ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
0               5849                0.0         NaN             360.0   
1               4583             1508.0       128.0             360.0   
2               3000                0.0        66.0             360.0   
3               2583             2358.0       120.0             360.0   
4               6000                0.0       141.0             360.0   
..               ...                ...         ...               ...   
609             2900                0.0        71.0             360.0   
610             4106                0.0        40.0             180.0   
611             8072              240.0       253.0             360.0   
612             7583                0.0       187.0             360.0   
613             4583                0.0       133.0             360.0   

     Credit_History Property_Area Loan_Status  
0               1.0         Urban           Y  
1               1.0         Rural           N  
2               1.0         Urban           Y  
3               1.0         Urban           Y  
4               1.0         Urban           Y  
..              ...           ...         ...  
609             1.0         Rural           Y  
610             1.0         Rural           Y  
611             1.0         Urban           Y  
612             1.0         Urban           Y  
613             0.0     Semiurban           N  

[614 rows x 13 columns]

### Langkah Kedua Praproses 
Data akan dibuat sebuah _copy_ dari _dataframe_ yang sudah di_load_ dari langkah awal, __data_copy__ ini akan dijadikan sebagai _dataframe_ yang akan dilakukan praproses data. Hal ini dilakukan untuk menghindari perubahan besar pada data utama yang dimiliki sehingga apabila ada kesalahan pada proses praproses akan dapat dengan mudah melihat kembali bentuk data awalnya seperti apa 

In [6]:
data_copy.isnull().sum()

Loan_ID               0
Gender               13
Married               3
Dependents           15
Education             0
Self_Employed        32
ApplicantIncome       0
CoapplicantIncome     0
LoanAmount           22
Loan_Amount_Term     14
Credit_History       50
Property_Area         0
Loan_Status           0
dtype: int64

### Langkah ketiga Mengatasi missing value
Dari hasil diatas dapat diketahui beberapa _missing value_ yang ada dalam dataset ini.

#### Penanganan _Missing Value_ Gender
Diketahui pada variable _gender_ terdapat 13 buah _missing value_. Dikarenakan gender merupakan data tipe kategorikal maka cara penanganannya adalah dengan mengisi _missing value_ menggunakan mode dari variable tersebut

In [7]:
mode = data_copy['Gender'].mode()[0]
data_copy['Gender'] = data_copy['Gender'].fillna(mode)

In [8]:
data_copy['Gender'].isnull().sum()

0

#### Penanganan _Missing Value_ Married
Diketahui pada variable _Married_ terdapat 3 buah _missing value_. Dikarenaka _Married_ merupakan data tipe kategorikal maka cara penanganannya adalah dengan mengisi _missing value_ menggunakan mode dari variable tersebut

In [9]:
mode = data_copy['Married'].mode()[0]
data_copy['Married'] = data_copy['Married'].fillna(mode)

In [10]:
data_copy['Married'].isnull().sum()

0

#### Penanganan _Missing Value_ Dependents
Diketahui pada variable _Dependents_ terdapat 15 buah _missing value_. Dikarenaka _Dependents_ merupakan data tipe kategorikal dalam bentuk numerik maka cara penanganannya adalah dengan mengisi _missing value_ menggunakan mode dari variable tersebut

In [11]:
mode = data_copy['Dependents'].mode()[0]
data_copy['Dependents'] = data_copy['Dependents'].fillna(mode)

data_copy['Dependents'].isnull().sum()

#### Penanganan _Missing Value_ Self employed
Diketahui pada variable _Self employed_ terdapat 32 buah _missing value_. Dikarenaka _Self employed_ merupakan data tipe kategorikal  maka cara penanganannya adalah dengan mengisi _missing value_ menggunakan mode dari variable tersebut

In [12]:
mode = data_copy['Self_Employed'].mode()[0]
data_copy['Self_Employed'] = data_copy['Self_Employed'].fillna(mode)

In [13]:
data_copy['Self_Employed'].isnull().sum()

0

#### Penanganan _Missing Value_ LoanAmount
Diketahui pada variable _LoanAmount_ terdapat 22 buah _missing value_. Dikarenaka _LoanAmount_ merupakan data tipe numerikal dalam maka cara penanganannya adalah dengan mengisi _missing value_ menggunakan mean dari variable tersebut

In [14]:
mean = data_copy['LoanAmount'].mean()
data_copy['LoanAmount'] = data_copy['LoanAmount'].fillna(int(mean))

#### Penanganan _Missing Value_ Loan Amount Term
Diketahui pada variable _Loan Amount Term_ terdapat 22 buah _missing value_. Dikarenaka _Loan Amount Term_ merupakan data tipe numerikal namu untuk waktu peminjaman ada aturan tertentu untuk berapa lama seseorang dapat melakukan peminjaman maka penanganan _missing value_  dengan mengisi _missing value_ menggunakan _mean_ kurang cocok diterapkan pada variable ini. Pengisian _missing value_ lebih cocok menggunakan _mode_

In [15]:
mode = data_copy['Loan_Amount_Term'].mode()[0]
data_copy['Loan_Amount_Term'] = data_copy['Loan_Amount_Term'].fillna(mode)

#### Penanganan _Missing Value_ Credit History
Diketahui pada variable _Credit History_ terdapat 32 buah _missing value_. Dikarenaka _Credit History_ merupakan data tipe kategorikal  maka cara penanganannya adalah dengan mengisi _missing value_ menggunakan mode dari variable tersebut

In [16]:
mode = data_copy['Credit_History'].mode()[0]
data_copy['Credit_History'] = data_copy['Credit_History'].fillna(mode)

In [17]:
data_copy.isnull().sum()

Loan_ID              0
Gender               0
Married              0
Dependents           0
Education            0
Self_Employed        0
ApplicantIncome      0
CoapplicantIncome    0
LoanAmount           0
Loan_Amount_Term     0
Credit_History       0
Property_Area        0
Loan_Status          0
dtype: int64

### Langkah 4 Konversi data
Dikarenakan model prediksi tidak dapat memprose data dengan _value string_, maka perlu dikonversi variable yang memiliki nilai _string_ menjadi numerikal.

Dalam kasus ini terdapat dua jenis konversi yang akan dilakukan yaitu :
    
    - Perubahan value kategorik yang memiliki 2 tipe (_binary_) menjadi 0 dan 1. Variable yang dirubah :
        - Married             = (Yes, No) Yes menjadi 1 dan No menjadi 0
        - Education           = (Graduate, Not Graduate) Graduate menjadi 1 dan Not Graduate menjadi 0
        - Self_Employed       = (Yes, No) Yes menjadi 1 dan No menjadi 0
        - Loan_Status         = (Y,N) Y menjadi 1 dan N menjadi 0
    
    - Perubahan value kategorik yang memiliki jenis lebih dari 2 atau bukan yes no, maka akan menggunakan dummy. Variable yang dirubah :
        - Gender              = (Male, Female)
        - Dependents          = (0–3)
        - Property_Area       = (Urban: Perkotaan, Semiurban: Semi perkotaan, Rural: Pedesaan)
        
    - Merubah Credit_History dari Float menjadi tipe int

In [18]:
data_copy['Self_Employed'] = data_copy['Self_Employed'].replace({'No': 0, 'Yes': 1})
data_copy['Married'] = data_copy['Married'].replace({'No': 0, 'Yes': 1})
data_copy['Education'] = data_copy['Education'].replace({'Not Graduate': 0, 'Graduate': 1})
data_copy['Loan_Status'] = data_copy['Loan_Status'].replace({'N': 0, 'Y': 1})
data_copy

Loan_ID  Gender  Married Dependents  Education  Self_Employed  \
0    LP001002    Male        0          0          1              0   
1    LP001003    Male        1          1          1              0   
2    LP001005    Male        1          0          1              1   
3    LP001006    Male        1          0          0              0   
4    LP001008    Male        0          0          1              0   
..        ...     ...      ...        ...        ...            ...   
609  LP002978  Female        0          0          1              0   
610  LP002979    Male        1         3+          1              0   
611  LP002983    Male        1          1          1              0   
612  LP002984    Male        1          2          1              0   
613  LP002990  Female        0          0          1              1   

     ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
0               5849                0.0       146.0             360.0   
1               4583             1508.0       128.0             360.0   
2               3000                0.0        66.0             360.0   
3               2583             2358.0       120.0             360.0   
4               6000                0.0       141.0             360.0   
..               ...                ...         ...               ...   
609             2900                0.0        71.0             360.0   
610             4106                0.0        40.0             180.0   
611             8072              240.0       253.0             360.0   
612             7583                0.0       187.0             360.0   
613             4583                0.0       133.0             360.0   

     Credit_History Property_Area  Loan_Status  
0               1.0         Urban            1  
1               1.0         Rural            0  
2               1.0         Urban            1  
3               1.0         Urban            1  
4               1.0         Urban            1  
..              ...           ...          ...  
609             1.0         Rural            1  
610             1.0         Rural            1  
611             1.0         Urban            1  
612             1.0         Urban            1  
613             0.0     Semiurban            0  

[614 rows x 13 columns]

In [19]:
data_copy = data_copy.join(pd.get_dummies(data_copy['Gender']))
data_copy = data_copy.join(pd.get_dummies(data_copy['Dependents']))
data_copy = data_copy.join(pd.get_dummies(data_copy['Property_Area']))
data_copy = data_copy.drop(columns = 'Gender')
data_copy = data_copy.drop(columns = 'Dependents')
data_copy = data_copy.drop(columns = 'Property_Area')
data_copy

Loan_ID  Married  Education  Self_Employed  ApplicantIncome  \
0    LP001002        0          1              0             5849   
1    LP001003        1          1              0             4583   
2    LP001005        1          1              1             3000   
3    LP001006        1          0              0             2583   
4    LP001008        0          1              0             6000   
..        ...      ...        ...            ...              ...   
609  LP002978        0          1              0             2900   
610  LP002979        1          1              0             4106   
611  LP002983        1          1              0             8072   
612  LP002984        1          1              0             7583   
613  LP002990        0          1              1             4583   

     CoapplicantIncome  LoanAmount  Loan_Amount_Term  Credit_History  \
0                  0.0       146.0             360.0             1.0   
1               1508.0       128.0             360.0             1.0   
2                  0.0        66.0             360.0             1.0   
3               2358.0       120.0             360.0             1.0   
4                  0.0       141.0             360.0             1.0   
..                 ...         ...               ...             ...   
609                0.0        71.0             360.0             1.0   
610                0.0        40.0             180.0             1.0   
611              240.0       253.0             360.0             1.0   
612                0.0       187.0             360.0             1.0   
613                0.0       133.0             360.0             0.0   

     Loan_Status  Female  Male  0  1  2  3+  Rural  Semiurban  Urban  
0              1       0     1  1  0  0   0      0          0      1  
1              0       0     1  0  1  0   0      1          0      0  
2              1       0     1  1  0  0   0      0          0      1  
3              1       0     1  1  0  0   0      0          0      1  
4              1       0     1  1  0  0   0      0          0      1  
..           ...     ...   ... .. .. ..  ..    ...        ...    ...  
609            1       1     0  1  0  0   0      1          0      0  
610            1       0     1  0  0  0   1      1          0      0  
611            1       0     1  0  1  0   0      0          0      1  
612            1       0     1  0  0  1   0      0          0      1  
613            0       1     0  1  0  0   0      0          1      0  

[614 rows x 19 columns]

In [20]:
data_copy['Credit_History'] = data_copy['Credit_History'].astype('int64')
data_copy.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 614 entries, 0 to 613
Data columns (total 19 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Loan_ID            614 non-null    object 
 1   Married            614 non-null    int64  
 2   Education          614 non-null    int64  
 3   Self_Employed      614 non-null    int64  
 4   ApplicantIncome    614 non-null    int64  
 5   CoapplicantIncome  614 non-null    float64
 6   LoanAmount         614 non-null    float64
 7   Loan_Amount_Term   614 non-null    float64
 8   Credit_History     614 non-null    int64  
 9   Loan_Status        614 non-null    int64  
 10  Female             614 non-null    uint8  
 11  Male               614 non-null    uint8  
 12  0                  614 non-null    uint8  
 13  1                  614 non-null    uint8  
 14  2                  614 non-null    uint8  
 15  3+                 614 non-null    uint8  
 16  Rural              614 non

## Langkah Pemabgian Data Training dan Testing
Pada langkah ini akan menggunakan _library_ __sklearn__ maka langkah pertama yang dilakukan adalah mengimport _library_ tersebut

In [21]:
from sklearn import model_selection #Library untuk proses prediksi
from sklearn.model_selection import KFold # Cross Validation
from sklearn.model_selection import cross_val_score # Mengambil nilai hasil tes
from sklearn.model_selection import cross_validate # Mengambil nilai hasil tes
from sklearn.model_selection import train_test_split # Repeated Hold Out
from sklearn.metrics import classification_report # Menampilkan hasil tes

## Pembagian Data Training dan Testing
Dalam pembagian dataset ini perlu ditentukan independent variable (y) dan dependent variable(x) yang akan digunakan. Dalam hal ini seluruh variable selain Loan ID dan Loan Status. Loan Status akan digunakan sebagai dependent variable(x)

In [22]:
fitur = ["Married", "Education", "Self_Employed", "ApplicantIncome", "CoapplicantIncome", "LoanAmount", "Loan_Amount_Term", "Credit_History", 
         "Female", "Male", "0", "1", "2", "3+", "Rural", "Semiurban", "Urban"]

x = data_copy[fitur]
y = data_copy['Loan_Status']
            
x_train, x_test, y_train, y_test = train_test_split(x, y, train_size=0.7, shuffle=True, random_state = 24)

## Membangun Model Prediksi
Dalam kasus ini metode yang akan digunakan dalam model adalah Naive Bayes. Kfold akan digunakan untuk membuat iterasi _testing_ yang akan dilakukan untuk mendapatkan nilai kinerja dari model.

In [23]:
from sklearn.naive_bayes import GaussianNB
naivebayes = GaussianNB()
kfold = model_selection.KFold()

## Melatih Model
Pada langkah ini model prediksi yang sudah dibuat akan di latih menggunakan data training yang sudah dibuat pada langkah sebelumnya. Training dilakukan dengan meggunakan _function_ .__fit()__

In [24]:
naivebayes.fit(x_train, y_train)

GaussianNB()

## Evaluasi Kinerja Model
Dalam tahap ini metriks yang digunakan untuk mengukur kinerja dari model tersebut adalah _Accuracy_ dan _f1-measure_ . Untuk mendapatkan nilai dari kedua metriks tersebut digunakan metode _Cross Validation_

In [25]:
score_test = naivebayes.score(x_test, y_test)
predict = naivebayes.predict(x_test)
print(classification_report(y_test, predict))

              precision    recall  f1-score   support

           0       0.84      0.47      0.61        57
           1       0.80      0.96      0.88       128

    accuracy                           0.81       185
   macro avg       0.82      0.72      0.74       185
weighted avg       0.82      0.81      0.79       185



In [26]:
scoring = ['accuracy', 'f1']
print(cross_val_score(naivebayes, x, y, cv=kfold, scoring="accuracy"))

result = cross_val_score(naivebayes, x, y, cv=kfold, scoring="accuracy")
print(result.mean())

[0.78861789 0.7398374  0.78861789 0.82113821 0.79508197]
0.7866586698653871


Nilai diatas merupakan hasi iterasi testing dari model yang sudah dibuat, dari hasil tersebut akan di rata-rata untuk mengetahui kinerja model secara menyeluruh sebesar 0.78 atau 78%

In [27]:
train_size = [0.9, 0.8, 0.75, 0.7, 2/3]
for t in train_size :
    naivebayes = GaussianNB()
    x_train, x_test, y_train, y_test = train_test_split(x, y, train_size=t, random_state=24)
    naivebayes.fit(x_train, y_train)
    score = naivebayes.score(x_test, y_test)
    print("Accuracy Naive Bayes with train size %.2f : %.3f%%" % (t, score*100))

Accuracy Naive Bayes with train size 0.90 : 83.871%
Accuracy Naive Bayes with train size 0.80 : 82.927%
Accuracy Naive Bayes with train size 0.75 : 83.766%
Accuracy Naive Bayes with train size 0.70 : 81.081%
Accuracy Naive Bayes with train size 0.67 : 80.488%


In [28]:
data_copy.corr()

Married  Education  Self_Employed  ApplicantIncome  \
Married            1.000000  -0.012304       0.004489         0.051708   
Education         -0.012304   1.000000       0.010383         0.140760   
Self_Employed      0.004489   0.010383       1.000000         0.127180   
ApplicantIncome    0.051708   0.140760       0.127180         1.000000   
CoapplicantIncome  0.075948   0.062290      -0.016100        -0.116605   
LoanAmount         0.147131   0.167041       0.115259         0.565621   
Loan_Amount_Term  -0.100912   0.073928      -0.033739        -0.046531   
Credit_History     0.010938   0.073658      -0.001550        -0.018615   
Loan_Status        0.091478   0.085884      -0.003700        -0.004710   
Female            -0.364569   0.045364       0.000525        -0.058809   
Male               0.364569  -0.045364      -0.000525         0.058809   
0                 -0.348175   0.036563      -0.088254        -0.092599   
1                  0.113853   0.013355       0.082044         0.040861   
2                  0.249547  -0.020822       0.032434        -0.034650   
3+                 0.132566  -0.055288       0.003278         0.156687   
Rural             -0.006805  -0.077520       0.022065         0.015829   
Semiurban          0.005845   0.039410       0.008710        -0.014246   
Urban              0.000546   0.034279      -0.030338        -0.000598   

                   CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
Married                     0.075948    0.147131         -0.100912   
Education                   0.062290    0.167041          0.073928   
Self_Employed              -0.016100    0.115259         -0.033739   
ApplicantIncome            -0.116605    0.565621         -0.046531   
CoapplicantIncome           1.000000    0.187863         -0.059383   
LoanAmount                  0.187863    1.000000          0.036486   
Loan_Amount_Term           -0.059383    0.036486          1.000000   
Credit_History              0.011134   -0.001412         -0.004705   
Loan_Status                -0.059187   -0.036345         -0.022549   
Female                     -0.082912   -0.107909          0.074030   
Male                        0.082912    0.107909         -0.074030   
0                          -0.008292   -0.147690          0.118163   
1                          -0.029769    0.062351         -0.088492   
2                           0.010016    0.019560         -0.010609   
3+                          0.041491    0.153195         -0.077273   
Rural                       0.005329    0.043185          0.034321   
Semiurban                  -0.027044   -0.008218          0.059141   
Urban                       0.022776   -0.033284         -0.094279   

                   Credit_History  Loan_Status    Female      Male         0  \
Married                  0.010938     0.091478 -0.364569  0.364569 -0.348175   
Education                0.073658     0.085884  0.045364 -0.045364  0.036563   
Self_Employed           -0.001550    -0.003700  0.000525 -0.000525 -0.088254   
ApplicantIncome         -0.018615    -0.004710 -0.058809  0.058809 -0.092599   
CoapplicantIncome        0.011134    -0.059187 -0.082912  0.082912 -0.008292   
LoanAmount              -0.001412    -0.036345 -0.107909  0.107909 -0.147690   
Loan_Amount_Term        -0.004705    -0.022549  0.074030 -0.074030  0.118163   
Credit_History           1.000000     0.540556 -0.009170  0.009170  0.020500   
Loan_Status              0.540556     1.000000 -0.017987  0.017987 -0.003044   
Female                  -0.009170    -0.017987  1.000000 -1.000000  0.148421   
Male                     0.009170     0.017987 -1.000000  1.000000 -0.148421   
0                        0.020500    -0.003044  0.148421 -0.148421  1.000000   
1                        0.009757    -0.038740  0.004466 -0.004466 -0.531373   
2                        0.007987     0.062384 -0.129953  0.129953 -0.528246   
3+                      -0.060473    -0.026123 -0.096319  0.096319 -0.358315   
Ru

## Kesimpulan
 Metode Naive Bayes dengan menggunakan variable Gender, Married, Dependents, Education, Self_Employed, ApplicantIncome, CoapplicantIncome, LoanAmount, Loan_Amount_Term, Credit_History, Property_Area sebagai variable independent dapat memprediksi kasus ini dengan cukup baik.
 
 Hasil prediksi dengan nilai tertinggi didapat dari model dengan _training size_ sebesar 0.67 dengan hasil akurasi 79.5%, dapat dilihat dari seluruh hasil prediksi dari model tersebut bahwa model ini dapat memprediksi dengan baik dikarenakan seluruh hasil prediksi memiliki tingkat akurasi lebih tinggi dari 70%